data/src/main/java/com/linkedin/data/codec/AbstractJacksonDataCodec.java

```diff
@@ -115,20 +115,11 @@ protected JsonGenerator createJsonGenerator(Writer out) throws IOException
 
   protected void writeObject(Object object, JsonGenerator generator) throws IOException
   {
-    try
+    try (Data.TraverseCallback callback = createTraverseCallback(generator))
     {
-      Data.TraverseCallback callback = createTraverseCallback(generator);
       Data.traverse(object, callback);
       generator.flush();
```

Comment:
Why do you need to flush here?
The behavior should be consistent: traverseCallback handles flush/close on the generator. And this should be documented on createTraverseCallback method

Reply:
generate close implementation has a condition on flush before stream close that why kept it here

Reply:
My point is about consistency between different codecs. Protobuf codec assumes the callback handles the flush, while this code has to call it explicitly. 

We should make it consistent and set the expectation that the traverse callback should handle flushing/closing the generator.
Then the codec's would just close the traverse callback.

Reply:
moved flush to callback close